process/smartContract/hooks/blockChainHook.go

```diff
@@ -394,6 +394,14 @@ func (bh *BlockChainHookImpl) LastTimeStamp() uint64 {
 	return 0
 }
 
+// LastTimeStampMs returns the timeStamp in milliseconds from the last committed block
+func (bh *BlockChainHookImpl) LastTimeStampMs() uint64 {
+	if !check.IfNil(bh.blockChain.GetCurrentBlockHeader()) {
+		return common.ConvertTimeStampSecToMs(bh.blockChain.GetCurrentBlockHeader().GetTimeStamp())
```

Comment:
ConvertTimeStampSecToMs looks like this
`
// ConvertTimeStampSecToMs will convert unix timestamp from seconds to milliseconds
// TODO: this has to be handled properly when round timestamp granularity will be changed to milliseconds
func ConvertTimeStampSecToMs(timeStamp uint64) uint64 {
	return timeStamp * 1000
}
`
With Supernova we'll handle this TODO properly and change in all places